### PR TITLE
feat: support configurables in typegen

### LINKED ISSUE #1014: Typegen - predicate typegen to support configurable constants
Now that predicates support configurable constants in #998, we should look to support them in typegen.

Predicates with configurable constants are instantiated like so:

https://github.com/FuelLabs/fuels-ts/blob/28c1cda969c2cf97519d9289798c18843dc3364b/packages/fuel-gauge/src/predicate-with-configurable.test.ts#L77

However the autogenerated predicate factory created using typegen doesn't offer any way of passing configurables:

https://github.com/FuelLabs/fuels-ts/blob/28c1cda969c2cf97519d9289798c18843dc3364b/packages/abi-typegen/src/templates/predicate/factory.hbs#L49-L62

We should be able to infer from the ABI that the predicate contains configurable constants and the types of them, and provide a way for them to be easily passed to the factory instance. This could be done by amending the `createInstance` signature should configurables be present.

### PR DESCRIPTION
**Pull request by danielbate**
Following the support of configurable constants in #998, this PR will infer the configurables from ABI and generate the appropriate type for all program types. Additionally, for predicates, this is also passed to the autogenerated predicate factory.

Closes #1014

### CODE REVIEW

File: packages/fuels/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "fuels",
-  "version": "0.43.1",
+  "version": "0.43.0",
```
Comment (danielbate) [inline packages/fuels/package.json:3]: Should not have been adjusted, will sort before live PR

File: packages/abi-typegen/src/templates/contract/dts.hbs [hunk 2]
```diff
@@ -40,6 +40,13 @@ export type {{structName}}Output{{typeAnnotations}} = { {{outputValues}} };
 {{/if}}
 {{/each}}
 
+{{#if formattedConfigurables}}
+export type {{capitalizedName}}Configurables = {
+  {{#each formattedConfigurables}}
+    {{configurableName}}: {{configurableType}};
+  {{/each}}
```
Comment (arboleya) [inline packages/abi-typegen/src/templates/contract/dts.hbs:47]: This fixes the double-indentation on the _rendered_ template.

The rendered fixtures will need to be adjusted accordingly.

```suggestion
{{#each formattedConfigurables}}
  {{configurableName}}: {{configurableType}};
{{/each}}
```

File: packages/abi-typegen/src/utils/makeConfigurable.test.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  }
+
+  it('should instantiate a configurable instance', () => {
+    const { type, rawAbiConfigurable } = mockAllDeps();
+    const types = [type];
+
+    const configurable = makeConfigurable({ types, rawAbiConfigurable });
+
+    expect(configurable).toBeInstanceOf(Configurable);
+    expect(configurable.name).toEqual('mockConfigurable');
```
Comment (arboleya) [inline packages/abi-typegen/src/utils/makeConfigurable.test.ts:48]:
```suggestion
    expect(configurable.name).toEqual(rawAbiConfigurable.name);
```

File: packages/abi-typegen/src/utils/parseConfigurables.test.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    const { type, rawAbiConfigurable, makeConfigurable } = mockAllDeps();
+    const rawAbiConfigurables = [rawAbiConfigurable];
+    const types = [type];
+
+    const configurables = parseConfigurables({ rawAbiConfigurables, types });
+
+    expect(makeConfigurable).toHaveBeenCalledTimes(1);
+
+    const configurable = configurables[0];
+    expect(configurable.name).toEqual('mockConfigurable');
```
Comment (arboleya) [inline packages/abi-typegen/src/utils/parseConfigurables.test.ts:61]:
```suggestion
    expect(configurable.name).toEqual(rawAbiConfigurable.name);
```

File: packages/fuels/package.json (hunk 1 above)
Comment (arboleya) [reply packages/fuels/package.json:3]: What do you mean? It's `0.43.1` on `master` ([here](https://github.com/FuelLabs/fuels-ts/blob/c8698f5f8d48ea8680f02719190334550616d6a3/packages/fuels/package.json#LL3C3-L3C23)).

**Review by arboleya — CHANGES_REQUESTED** [review]
Great work on this! 🤘 

Left a few trivial observations.

**Review by arboleya — APPROVED** [review]
Solid. 💯

**Review by Torres-ssf — APPROVED** [review]

**Pull request merged**